R/framework_ebp.R

```diff
@@ -90,38 +106,41 @@ framework_ebp <- function(fixed, pop_data, pop_domains, smp_data, smp_domains,
   )
 
   indicator_list <- list(
-    fast_mean = function(y, threshold) {
-      t(mean(y))
+    fast_mean = function(y, w, threshold) {
+      t(weighted.mean(y, w))
     },
-    hcr = function(y, threshold) {
-      t(mean(y < threshold))
+    hcr = function(y, w, threshold) {
+       t(weighted.mean(y < threshold, w))
     },
-    pgap = function(y, threshold) {
-      t(mean((y < threshold) * (threshold - y) / threshold))
+    pgap = function(y, w, threshold) {
+      sum((1 - (y[y < threshold]/threshold)) * w[y < threshold])/sum(w)
     },
-    gini = function(y, threshold) {
-      n <- length(y)
-      y <- sort(y)
-      G <- sum(y * 1L:n)
-      G <- 2 * G / sum(y) - (n + 1L)
-      G <- t(G / n)
-      return(G)
+    gini = function(y, w, threshold) {
+        n <- length(y)
+        w <- w[order(y)]
+        y <- sort(y)
+        auc <- sum((cumsum(c(0, (y * w)[1:(n-1)])) + ((y * w) / 2)) * w)
+        auc <- (auc / sum(w)) / sum((y * w))
+        G <- 1 - (2* auc)
+        return(G)
     },
-    qsr = function(y, threshold) {
-      weights <- rep.int(1, length(y))
-      quant14 <- wtd.quantile(
-        x = y,
-        weights = weights,
-        probs = c(0.2, 0.8)
-      )
-
+    qsr = function(y, w, threshold) {
+      if(length(unique(w)) == 1 & 1 %in% unique(w)){
+        quant14 <- quantile(x = y, probs = c(0.2, 0.8))
```

Comment:
Does this now still return the same as before? Before it seems that we used wtd.quantile for the qsr with 1 as weights, when we change this now to quantile, shouldnt it give a difference?

Reply:
geändert und getestet, führt bei allen Beispielen zu den selben Ergebnissen